packages/flutter/test/widgets/selection_container_test.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('SelectionContainer', () {
+
+    Future<void> pumpContainer(WidgetTester tester, Widget child) async {
+      await tester.pumpWidget(
+        DefaultSelectionStyle(
+          selectionColor: Colors.red,
+          child: child,
+        ),
+      );
+    }
+
+    testWidgets('updates its registrar and delegate based on the number of selectables', (WidgetTester tester) async {
+      final TestSelectionRegistrar registrar = TestSelectionRegistrar();
+      final TestContainerDelegate delegate = TestContainerDelegate();
+      await pumpContainer(
+        tester,
+        SelectionContainer(
+          registrar: registrar,
+          delegate: delegate,
+          child: Column(
+            children: const <Widget>[
+              Text('column1', textDirection: TextDirection.ltr),
+              Text('column2', textDirection: TextDirection.ltr),
+              Text('column3', textDirection: TextDirection.ltr),
+            ],
+          ),
+        ),
+      );
+      expect(registrar.selectables.length, 1);
+      expect(delegate.selectables.length, 3);
```

Comment:
It's not immediately clear to me why these aren't both 3...

Reply:
`registrar` will be what selection container register to, 
`delegate` will be what the selectable (3 text widgets) in the subtree register too

Reply:
Got it, thanks.